crates/astria-sequencer/src/mempool/transactions_container.rs

```diff
@@ -200,17 +324,45 @@ pub(super) struct ParkedTransactionsForAccount<const MAX_TX_COUNT: usize> {
 
 impl<const MAX_TX_COUNT: usize> ParkedTransactionsForAccount<MAX_TX_COUNT> {
     /// Returns contiguous transactions from front of queue starting from target nonce, removing the
-    /// transactions in the process.
+    /// transactions in the process. Will only return transactions if their cost is covered by the
+    /// `available_balances`.
     ///
     /// Note: this function only operates on the front of the queue. If the target nonce is not at
-    /// the front, an error will be logged and nothing will be returned.
-    fn pop_front_contiguous(
+    /// the front, nothing will be returned.
```

Comment:
why pass in the target nonce if the function only works at the front of the queue?

Reply:
The target nonce is the next nonce that pending is expecting, I can add that to the documentation. It's not guaranteed that parked will contain a transaction with the target nonce and I was letting this function internalize that check instead of doing it explicitly. I can make it explicit if you'd prefer. 

There shouldn't be the scenario where parked and pending have overlapping nonces as we only add things to parked if their nonce isn't already in pending.